### PR TITLE
`Development`: Move methods from UtilService to Factory in server tests

### PR DESCRIPTION
**Pull request by laadvo**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

When refactoring `DatabaseUtilService` some methods responsible for creating objects (with no database interractions) were put into the new UtilServices instead of Factories. This is fixed in this PR.

### Description

Mostly just moving methods without changing anything from UtilService into the corresponding Factory. Otherswise I also removed some helper methods that are no longer used. In `ConversationUtilService.java` I had to change a mehod call, so that the ` userUtilService.getUserByLoginWithoutAuthorities` is still called from the `ConversationUtilService` and not from the `ConversationFactory`.

### Review Progress

Since it is mostly just moving methods, it should be enough to just check the `ConversationUtilService` and `ConversationFactory`.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseBitbucketBambooIntegrationTest.java [hunk 1]
```diff
@@ -373,7 +370,7 @@ void importExerciseFromFile_valid_Exercise_importSuccessful(ProgrammingLanguage
         programmingExerciseTestService.importFromFile_validExercise_isSuccessfullyImported(language);
     }
 
-    @Test
+    @RepeatedTest(1000)
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseBitbucketBambooIntegrationTest.java:373]: 👀

**Review by DominikRemo — CHANGES_REQUESTED** [review]

File: src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseBitbucketBambooIntegrationTest.java (hunk 1 above)
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseBitbucketBambooIntegrationTest.java]: thanks for noticing 😅

**Review by lennart-keller — DISMISSED** [review]
Code lgtm :)

File: src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseBitbucketBambooIntegrationTest.java [hunk 2]
```diff
@@ -14,10 +14,7 @@
 import java.util.stream.Stream;
 
 import org.json.simple.parser.JSONParser;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Disabled;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseBitbucketBambooIntegrationTest.java:17]:
```suggestion
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
```

**Review by DominikRemo — APPROVED** [review]
Code LGTM

**Review by DominikRemo — CHANGES_REQUESTED** [review]
I was to fast with my last approval ^^"
Merging the changes from develop seems to have broken something within the tests so they fail now

**Review by terlan98 — COMMENTED** [review]
Code LGTM but the tests are failing, as mentioned by Dominik

File: src/test/java/de/tum/in/www1/artemis/exam/ExamIntegrationTest.java [hunk 3]
```diff
@@ -111,6 +111,9 @@ class ExamIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTe
     @Autowired
     private ExamUserRepository examUserRepository;
 
+    @Autowired
+    private QuizExerciseUtilService quizExerciseUtilService;
+
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/exam/ExamIntegrationTest.java:116]: Unused according to teamscale

**Review by DominikRemo — CHANGES_REQUESTED** [review]
Code looks good to me, left one comment

**Review by DominikRemo — APPROVED** [review]
LGTM

Thanks for addressing my suggestion

**Review by lennart-keller — APPROVED** [review]
Recent code changes look good to me

**Pull request merged**